tools/ci/build-test-ubuntu-with-latest-clang.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# This script builds iceoryx_hoofs und iceoryx_posh and executes all tests
+
+set -e
+
+DIR="$(cd -P "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+msg() {
+    printf "\033[1;32m%s: %s\033[0m\n" ${FUNCNAME[1]} "$1"
+}
+
+# go to repo root
+ROOT=${DIR}/../..
+cd ${ROOT}
+
+msg "installing build dependencies"
+sudo apt-get update && sudo apt-get install -y libacl1-dev libncurses5-dev
+
+msg "installing latest stable clang"
+sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
+
+msg "creating local test users and groups for testing access control"
+sudo ./tools/add_test_users.sh
+
+msg "compiler versions:
+$(clang-12 --version)"
```

Comment:
With line 36 this breaks immediately with the release of clang-13. We either need to explicitly install the latest version in the ubuntu repositories, which requires to specify the version number. Or we need to find out which version is installed from `https://apt.llvm.org/llvm.sh`

Reply:
done